ProcessMaker/Models/EncryptedData.php

```diff
@@ -46,10 +49,59 @@ class EncryptedData extends ProcessMakerModel
      * @param string $userId
      * @param string $screenId
      * @param string $fieldName
+     *
+     * @throws \Illuminate\Validation\ValidationException
      */
     public static function checkUserPermission($userId, $screenId, $fieldName)
     {
-        // To Do: Complete check if the user is assigned to the encrypted field in the screen
+        // Get screen fields
+        $screen = Screen::find($screenId);
+        $fields = $screen->getFieldsAttribute()->toArray();
+
+        // Search encrypted fields
+        $encryptedFields = array_filter($fields, function ($field) use ($fieldName) {
+            return $field->field === $fieldName;
+        });
+
+        // Use the first definition
+        $encryptedField = array_pop($encryptedFields) ?? null;
+
+        // Run validations
+        if (is_null($encryptedField)) {
+            throw ValidationException::withMessages([
+                'encrypted_field_not_exists' => 'Not exists a field with name "' . $fieldName . '".',
+            ]);
+        }
+        if (is_null($encryptedField->encryptedConfig)) {
+            throw ValidationException::withMessages([
+                'encrypted_field_config_empty' => 'Config related to encryption is not found for this field.',
+            ]);
+        }
+        if (is_null($encryptedField->encryptedConfig['encrypted'])) {
+            throw ValidationException::withMessages([
+                'encrypted_field_encryption_disabled' => 'Config related to encryption is not enabled for this field.',
+            ]);
+        }
+        if (empty($encryptedField->encryptedConfig['assignments']['users']) && empty($encryptedField->encryptedConfig['assignments']['groups'])) {
+            throw ValidationException::withMessages([
+                'encrypted_field_assignments_empty' => 'Config related to assignments is missing for this field.',
+            ]);
+        }
+
+        // Get groups and users from config
+        $groupsAssigned = $encryptedField->encryptedConfig['assignments']['groups'];
+        $usersAssigned = $encryptedField->encryptedConfig['assignments']['users'];
+
+        // Get consolidated users list
+        $process = new Process();
+        $process->getConsolidatedUsers($groupsAssigned, $usersAssigned);
```

Comment:
For better readability consider to use:
```
$usersAssigned = $process->getConsolidatedUsers($groupsAssigned, $usersAssigned);
```
even if the $usersAssigned is passed by reference.